Enabled custom date masks and support for non-Date models

pavelpykhtin opened the pull request:
Hi.

I think it could be useful to have a way to customize actual mask format.
Also in our project we have to treat model value as a string. Yo achieve this 'parse' flag was added. Perhaps this functionality will be useful for someone else.

assisrafael commented:
Thank you for the pull request.
However I can not accept it until all tests pass.
Can you check the broken tests?

pavelpykhtin commented:
Hi

I checked and fixed broken tests. However I'm not sure that it was related to my changes (maybe I'm missing something).

Problem was related to the default formatting being applied by ngBind to a Date values. Expected date value were converted to string and compared with the value of the span bound to the model. Seems that ngBind uses format different from the format used by Date.toString(). [Here](https://gyazo.com/3a2e8387ceffec44c1514937dda5b180) is a screenshot of the test form, look at the 'Model value' field.
To fix this I modified check to compare expected value with the actual model value.

assisrafael commented:
Thank you!
